README.md

```diff
@@ -134,6 +134,65 @@ This is what each parameter means:
 | `operation_name` | The name of the operation that created the task                     |
 | `task_id`        | The MOID (Managed Object Reference ID) that identifies target task  |
 
+## Guest Operations
+
+The Guest Operations API allows for direct file and process manipulation inside a virtual machine.
+Consider the use case where you want to run a PowerShell script inside a virtual machine.  Here are the steps you would take:
+
+1. Create a PowerShell script in your own pack (for example "mypack/scripts/Return.ps1" - see below)
+2. Execute the vsphere.guest_script_run workflow.
+3. Examine the result - if the script returns a non-zero exit code, the action will fail; otherwise exit_code, stdout, and stderr are available.
```

Comment:
Actually it will grab all of those regardless.  I will update the comment.

Reply:
@nmaludy so in this case there's a workflow (guest_script_run) that publishes ctx().script_exit_code (to 1), and if the end of an Orquesta workflow has this:
```
    next:
      - when: <% ctx().script_exit_code != 0 %>
        do: fail
```
The top level workflow properly fails, but the output for the overall workflow shows what subtask failed:
```
{
  "output": null,
  "errors": [
    {
      "message": "Execution failed. See result for details.",
      "result": {
        "result": "None",
        "exit_code": 1,
        "stderr": "/opt/stackstorm/st2/lib/python2.7/site-packages/cryptography/hazmat/primitives/constant_time.py:26: CryptographyDeprecationWarning: Support for your Python version is deprecated. The next version of cryptography will remove support. Please upgrade to a 2.7.x release that supports hmac.compare_digest as soon as possible.\n  utils.DeprecatedIn23,\n",
        "stdout": "(vim.vm.guest.ProcessManager.ProcessInfo) {\n   dynamicType = <unset>,\n   dynamicProperty = (vmodl.DynamicProperty) [],\n   name = '\"CMD.EXE',\n   pid = 788L,\n   owner = 'Administrator',\n   cmdLine = '\"CMD.EXE\" /C NET SHOW 1> stdout 2> stderr',\n   startTime = 2018-10-09T17:40:22Z,\n   endTime = 2018-10-09T17:40:23Z,\n   exitCode = 1\n}\n"
      },
      "task_id": "process_wait"
    },
    {
      "message": "Execution failed. See result for details.",
      "task_id": "fail__2"
    }
  ]
}
```
Without this "next" at the end of the workflow, the top level workflow "succeeds" because the transition from process_wait to getting stdout is unconditional.... however the workflow properly promotes script_exit_code, script_stdout, and script_stderr from the ctx() to the output dict.  In this case it's the output of "NET SHOW" (which is not a valid command).

So this looks like an issue in how Orquesta handles task failures.  It doesn't appear to attempt to honor the output fields when the workflow fails.